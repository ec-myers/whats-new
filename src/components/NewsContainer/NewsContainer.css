  .newsContainer {
    padding: 30px;
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 20px;
    overflow: scroll;
  }