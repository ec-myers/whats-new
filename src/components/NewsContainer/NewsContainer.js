import React from 'react';
import './NewsContainer.css';
import NewsArticle from '../NewsArticle/NewsArticle';

const NewsContainer = props => {
  const newsData = props.news.map((article, i) => {
    return <NewsArticle key={i}
      headline={article.headline}
      img={article.img}
      url={article.url}
      description={article.description}
      />
  })

  return (
    <section className="newsContainer">
      {newsData}
    </section>
  )
}
export default NewsContainer;