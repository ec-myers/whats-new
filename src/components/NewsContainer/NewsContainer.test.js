import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {
  let news = [{
    id: 1,
    headline: 'India\'s Lost Moon Lander Is Somewhere in This Photo',
    img: 'https://cdn.mos.cms.futurecdn.net/FsUeVGmY89BWeb36U8cUj5-650-80.jpg',
    description: 'The image was captured by NASA\'s Lunar Reconnaissance Orbiter (LRO) on Sept. 17 as it flew over the targeted landing site of India\'s Chandrayaan-2 mission.',
    url: 'https://www.livescience.com/india-moon-landing-site-nasa-photo.html'
  },
  {
    id: 1,
    headline: 'Man dies after getting infection linked to shellfish, North Carolina officials say',
    img: 'https://www.newsobserver.com/entertainment/restaurants/o6w6n7/picture184859643/alternates/FREE_768/MAIN%20OPTION%202:%20OYSTERS',
    description: 'A North Carolina man has died after getting an infection often associated with shellfish, state health officials say.',
    url: 'https://www.bnd.com/news/nation-world/national/article235549897.html'
  }];

  it('should match the snapshot', () => {
    const wrapper = shallow(<NewsContainer news={news}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});