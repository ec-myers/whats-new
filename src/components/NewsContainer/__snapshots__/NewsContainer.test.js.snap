// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot 1`] = `
<section
  className="newsContainer"
>
  <NewsArticle
    description="The image was captured by NASA's Lunar Reconnaissance Orbiter (LRO) on Sept. 17 as it flew over the targeted landing site of India's Chandrayaan-2 mission."
    headline="India's Lost Moon Lander Is Somewhere in This Photo"
    img="https://cdn.mos.cms.futurecdn.net/FsUeVGmY89BWeb36U8cUj5-650-80.jpg"
    key="0"
    url="https://www.livescience.com/india-moon-landing-site-nasa-photo.html"
  />
  <NewsArticle
    description="A North Carolina man has died after getting an infection often associated with shellfish, state health officials say."
    headline="Man dies after getting infection linked to shellfish, North Carolina officials say"
    img="https://www.newsobserver.com/entertainment/restaurants/o6w6n7/picture184859643/alternates/FREE_768/MAIN%20OPTION%202:%20OYSTERS"
    key="1"
    url="https://www.bnd.com/news/nation-world/national/article235549897.html"
  />
</section>
`;
